src/node/internal/internal_http_server.ts

```diff
@@ -235,11 +240,18 @@ export class Server
     }
     incoming._addHeaderLines(headers, headers.length);
 
-    // TODO(soon): It would be useful if there was a way to expose request.cf properties.
     incoming.method = request.method;
     incoming._stream = request.body;
 
-    const response = new this[kServerResponse](incoming);
+    // We provide a way for users to access to the Cloudflare-specific
+    // request properties, such as `cf` for accessing Cloudflare-specific request metadata.
+    if ('cf' in request) {
+      incoming.cf = request.cf as Record<string, unknown>;
+    }
```

Comment:
This should have a test

Reply:
The test can be as simple as checking `'cf' in incoming`.

Also, the `if ('cf' in request)` guard is likely unnecessary.

Reply:
This will throw a typescript error.

Property 'cf' does not exist on type 'Request'. (ts 2339)